File: Games/Dennis/src/de/hsh/dennis/model/KeyLayout.java
package de.hsh.dennis.model;

import javafx.scene.input.KeyCode;

//fixed key bindings for DDosDefender
public class KeyLayout {

    public static class Movement {

        public static class Custom {
            //UP
            public static KeyCode UP = KeyCode.UP;
            public static KeyCode UP_ALT = KeyCode.W;

            //hit left
            public static KeyCode LEFT = KeyCode.LEFT;
            public static KeyCode LEFT_ALT = KeyCode.A;

            //Crouch
            public static KeyCode DOWN = KeyCode.DOWN;
            public static KeyCode DOWN_ALT = KeyCode.S;

            //hit right
            public static KeyCode RIGHT = KeyCode.RIGHT;
            public static KeyCode RIGHT_ALT = KeyCode.D;
        }

        static class Standard {
            //UP
            static final KeyCode UP = KeyCode.UP;
            static final KeyCode UP_ALT = KeyCode.W;

            //hit left
            static final KeyCode LEFT = KeyCode.LEFT;
            static final KeyCode LEFT_ALT = KeyCode.A;

            //Crouch
            static final KeyCode DOWN = KeyCode.DOWN;
            static final KeyCode DOWN_ALT = KeyCode.S;

            //hit right
            static final KeyCode RIGHT = KeyCode.RIGHT;
            static final KeyCode RIGHT_ALT = KeyCode.D;
        }

        public void movementToDefault() {
            Custom.UP = Standard.UP;
            Custom.UP_ALT = Standard.UP_ALT;
            Custom.LEFT = Standard.LEFT;
            Custom.LEFT_ALT = Standard.LEFT_ALT;
            Custom.DOWN = Standard.DOWN;
            Custom.DOWN_ALT = Standard.DOWN_ALT;
            Custom.RIGHT = Standard.RIGHT;
            Custom.RIGHT_ALT = Standard.RIGHT_ALT;

        }
    }

    //open break menu
    public static class Control {
        public static final KeyCode BREAK = KeyCode.ESCAPE;
        public static final KeyCode BREAK_ALT = KeyCode.P;
    }
}

File: Games/Dennis/src/de/hsh/dennis/model/KeyLayout.kt
package de.hsh.dennis.model

import javafx.scene.input.KeyCode

//fixed key bindings for DDosDefender
class KeyLayout {
    class Movement {
        object Custom {
            //UP
            var UP = KeyCode.UP
            var UP_ALT = KeyCode.W

            //hit left
            var LEFT = KeyCode.LEFT
            var LEFT_ALT = KeyCode.A

            //Crouch
            var DOWN = KeyCode.DOWN
            var DOWN_ALT = KeyCode.S

            //hit right
            var RIGHT = KeyCode.RIGHT
            var RIGHT_ALT = KeyCode.D
        }

        internal object Standard {
            //UP
            val UP = KeyCode.UP
            val UP_ALT = KeyCode.W

            //hit left
            val LEFT = KeyCode.LEFT
            val LEFT_ALT = KeyCode.A

            //Crouch
            val DOWN = KeyCode.DOWN
            val DOWN_ALT = KeyCode.S

            //hit right
            val RIGHT = KeyCode.RIGHT
            val RIGHT_ALT = KeyCode.D
        }

        fun movementToDefault() {
            Custom.UP = Standard.UP
            Custom.UP_ALT = Standard.UP_ALT
            Custom.LEFT = Standard.LEFT
            Custom.LEFT_ALT = Standard.LEFT_ALT
            Custom.DOWN = Standard.DOWN
            Custom.DOWN_ALT = Standard.DOWN_ALT
            Custom.RIGHT = Standard.RIGHT
            Custom.RIGHT_ALT = Standard.RIGHT_ALT
        }
    }

    //open break menu
    object Control {
        @JvmField
        val BREAK = KeyCode.ESCAPE
        @JvmField
        val BREAK_ALT = KeyCode.P
    }
}
